core/src/main/java/com/aem/demo/core/services/impl/SearchServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package com.aem.demo.core.services.impl;
+
+import com.aem.demo.core.models.NewsCardModel;
+import com.aem.demo.core.services.SearchService;
+import com.aem.demo.core.utils.ResolverUtil;
+import com.day.cq.search.PredicateGroup;
+import com.day.cq.search.Query;
+import com.day.cq.search.QueryBuilder;
+import com.day.cq.search.result.SearchResult;
+import com.day.cq.wcm.api.Page;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.sling.api.resource.LoginException;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceResolverFactory;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import javax.jcr.Session;
+import java.util.*;
+
+import static com.adobe.aemds.guide.utils.JcrResourceConstants.CQ_PAGE;
+
+@Slf4j
+@Component(service = SearchService.class, immediate = true)
+public class SearchServiceImpl implements SearchService {
+
+    private static final String ROOT_PATH = "/content/aemtraining/language-masters";
+
+
+    @Reference
+    QueryBuilder queryBuilder;
+
+    @Reference
+    ResourceResolverFactory resourceResolverFactory;
+
+    public Map<String, String> createTextSearchQuery(String searchText) {
```

Comment:
This method must not be `public`. This is not intended to be called from outside the class.